controllers/users/register_v2.js

```diff
@@ -28,47 +28,87 @@ const registerV2 = async (req, res) => {
-      const locations = await LocationFunction.getAllLocationByIds(Locations, local_community, t);
-
-      return {
-        user,
-        anonymousUser,
-        topics: topicRegistered,
-        locations
-      };
-    });
-  } catch (e) {
-    console.log('error on sql transaction', e);
-    return res.status(500).json({
-      status: 'error on sql transaction',
-      code: 500,
-      message: e
-    });
+  if (isUnverifiedUser) {
+    let updatedUser;
+    try {
+      await checkUser.update(users);
+      updatedUser = await checkUser.save();
+    } catch (e) {
+      return res.status(500).json({
+        status: 'error on sql transaction',
+        code: 500,
+        message: e
+      });
+    }
+
+    const topicRegistered = await TopicFunction.findAllByTopicIds(Topics, topics);
+    const locations = await LocationFunction.getAllLocationByIds(Locations, local_community);
+
+    insertedObject = {
+      user: updatedUser,
+      anonymousUser: await UsersFunction.findAnonymousUserId(User, updatedUser.user_id, {
+        raw: false
+      }),
+      topics: topicRegistered,
+      locations
+    };
+  } else {
+    /**
+     * Inserting data to Postgres DB
+     */
+    try {
+      insertedObject = await sequelize.transaction(async (t) => {
+        const user = await UsersFunction.register(User, users, t);
+        const anonymousUser = await UsersFunction.registerAnonymous(User, user?.user_id, t);
+
+        const topicRegistered = await TopicFunction.findAllByTopicIds(Topics, topics, t, true);
+        const locations = await LocationFunction.getAllLocationByIds(Locations, local_community, t);
+
+        return {
+          user,
+          anonymousUser,
+          topics: topicRegistered,
+          locations
+        };
+      });
+    } catch (e) {
+      console.log('error on sql transaction', e);
+      return res.status(500).json({
+        status: 'error on sql transaction',
+        code: 500,
+        message: e
+      });
+    }
+    /**
+     * Inserting data to Postgres DB (END)
+     */
   }
-  /**
-   * Inserting data to Postgres DB (END)
-   */
 
   /**
    * Creating User to Getstream
    */
   try {
-    token = await BetterSocialCore.user.createUser(insertedObject?.user);
-    anonymousToken = Getstream.core.createToken(
-      insertedObject?.anonymousUser?.user_id?.toLowerCase()
-    );
-    await BetterSocialCore.user.createAnonymousUser(insertedObject?.anonymousUser);
+    if (isUnverifiedUser) {
+      token = Getstream.core.createToken(insertedObject?.user?.user_id);
+      anonymousToken = Getstream.core.createToken(
+        insertedObject?.anonymousUser?.user_id?.toLowerCase()
+      );
+    } else {
+      token = await BetterSocialCore.user.createUser(insertedObject?.user);
+      anonymousToken = Getstream.core.createToken(
+        insertedObject?.anonymousUser?.user_id?.toLowerCase()
+      );
+      await BetterSocialCore.user.createAnonymousUser(insertedObject?.anonymousUser);
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In this section, you are creating tokens for both verified and unverified users. However, there is no error handling if the token creation fails. Consider adding a try-catch block around each token creation process to handle any potential errors.

```diff
-      token = Getstream.core.createToken(insertedObject?.user?.user_id);
-      anonymousToken = Getstream.core.createToken(
-        insertedObject?.anonymousUser?.user_id?.toLowerCase()
-      );
+      try {
+        token = Getstream.core.createToken(insertedObject?.user?.user_id);
+        anonymousToken = Getstream.core.createToken(
+          insertedObject?.anonymousUser?.user_id?.toLowerCase()
+        );
+      } catch (e) {
+        console.error('Error creating tokens:', e);
+        return res.status(500).json({
+          status: 'error',
+          code: 500,
+          message: 'There was an error processing your request. Please try again later.'
+        });
+      }
```

112-end:
The error handling in this section could also be improved by logging the error and returning a more user-friendly message.

```diff
-    console.log('error on inserting user to getstream creating', e);
-    return res.status(500).json({
-      status: 'error on inserting user to getstream creating',
-      code: 500,
-      message: e
-    });
+    console.error('Error on inserting user to Getstream:', e);
+    return res.status(500).json({
+      status: 'error',
+      code: 500,
+      message: 'There was an error processing your request. Please try again later.'
+    });
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->